### PR TITLE
Deprecate write_raw_bids()'s events_data parameter in favor of "events"

### PR DESCRIPTION
**Pull request by hoechenberger**
For more consistency with MNE-Python.

This is WIP.

Raises a FutureWarning if events_data is used. The idea is to have a very long deprecation cycle here (2, 3 releases?)

WDYT?

Merge checklist
---------------

Maintainer, please confirm the following before merging.
If applicable:

- [ ] All comments are resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] New contributors have been added to [CITATION.cff](https://github.com/mne-tools/mne-bids/blob/main/CITATION.cff)
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: mne_bids/write.py [hunk 1]
```diff
@@ -1244,11 +1244,23 @@ def make_dataset_description(*, path, name, hed_version=None,
...
-                   anonymize=None, format='auto', symlink=False,
-                   empty_room=None, allow_preload=False,
-                   montage=None, acpc_aligned=False,
-                   overwrite=False, verbose=None):
+def write_raw_bids(
+    raw,
+    bids_path,
+    *,
+    events=None,
+    events_data=None,
```
Comment (hoechenberger) [inline mne_bids/write.py:1252]: For now we accept both parameters. Passing events_data will emit a FutureWarning.

File: mne_bids/write.py [hunk 2]
```diff
@@ -1483,6 +1499,21 @@ def write_raw_bids(raw, bids_path, events_data=None, event_id=None,
     mne.events_from_annotations
 
     """
+    if events_data and events:
```
Comment (hoechenberger) [inline mne_bids/write.py:1502]:
```suggestion
    if events_data is not None and events is not None:
```

**Comment by hoechenberger** [comment]
Doc builds are failing with:

```
ImportError: cannot import name 'sphinx_compatibility' from 'sphinx_gallery' (/home/circleci/mne_bids_env/lib/python3.9/site-packages/sphinx_gallery/__init__.py)
```

File: mne_bids/commands/mne_bids_raw_to_bids.py [hunk 3]
```diff
@@ -81,7 +83,7 @@ def run():
         line_freq = None if opt.line_freq == "None" else opt.line_freq
         raw.info['line_freq'] = line_freq
     write_raw_bids(raw, bids_path, event_id=opt.event_id,
-                   events_data=opt.events_data, overwrite=opt.overwrite,
+                   events=opt.events, overwrite=opt.overwrite,
```
Comment (agramfort) [inline mne_bids/commands/mne_bids_raw_to_bids.py:86]: This will break backward compat if you don’t pass events_data no?

File: mne_bids/write.py [hunk 4]
```diff
@@ -1244,11 +1244,23 @@ def make_dataset_description(*, path, name, hed_version=None,
...
 @verbose
-def write_raw_bids(raw, bids_path, events_data=None, event_id=None,
-                   anonymize=None, format='auto', symlink=False,
-                   empty_room=None, allow_preload=False,
-                   montage=None, acpc_aligned=False,
-                   overwrite=False, verbose=None):
+def write_raw_bids(
+    raw,
+    bids_path,
+    *,
```
Comment (agramfort) [inline mne_bids/write.py:1250]: This is not backward compatible. I would add events_data as the end of the parameter list like done in the good old days

File: mne_bids/write.py [hunk 5]
```diff
@@ -1483,6 +1499,21 @@ def write_raw_bids(raw, bids_path, events_data=None, event_id=None,
...
     """
+    if events_data is not None and events is not None:
+        raise ValueError('Only one of events and events_data can be passed.')
+
+    if events_data is not None:
+        warn(
+            message='The events_data parameter has been deprecated in favor '
+                    'the new events parameter, to ensure better consistency  '
+                    'with MNE-Python. The events_data parameter will be '
+                    'removed in a future release of MNE-BIDS. Please use the '
```
Comment (agramfort) [inline mne_bids/write.py:1510]: When you deprecate say which version will drop support

**Review by agramfort — COMMENTED** [review]
Ok for me. I had thought about this in the past but it was loo priority for me. 

@sappelhoff what are you thoughts on this?

File: mne_bids/write.py (hunk 5 above)
Comment (hoechenberger) [reply mne_bids/write.py:1510]: We need to decide upon that :)

File: mne_bids/commands/mne_bids_raw_to_bids.py (hunk 3 above)
Comment (hoechenberger) [reply mne_bids/commands/mne_bids_raw_to_bids.py]: Addressed in [f6de206](https://github.com/mne-tools/mne-bids/pull/1054/commits/f6de2065f60f52e4cf4fc03606da636cdcd831f8)

File: mne_bids/write.py (hunk 4 above)
Comment (hoechenberger) [reply mne_bids/write.py]: Addressed in [65f181d](https://github.com/mne-tools/mne-bids/pull/1054/commits/65f181d949c23d75d01d8a8a1859b0ba94947f57), better now..? I'm still introducing kw-only arguments, but now only after the 4th parameter

File: mne_bids/write.py (hunk 5 above)
Comment (hoechenberger) [reply mne_bids/write.py:1510]: I'd say 0.14!

**Review by agramfort — APPROVED** [review]
@sappelhoff I let you have the 2nd look

**Review by sappelhoff — APPROVED** [review]
I think that this improvement in consistency is great, thanks for the work!

**Comment by agramfort** [comment]
thx @hoechenberger

**Pull request merged**